common/search/src/main/scala/uk/ac/wellcome/platform/api/rest/SingleWorkDirectives.scala

```diff
@@ -0,0 +1,47 @@
+package uk.ac.wellcome.platform.api.rest
+
+import akka.http.scaladsl.model.StatusCodes.Found
+import akka.http.scaladsl.server.Route
+import weco.api.search.elasticsearch.ElasticsearchError
+import weco.catalogue.internal_model.work.WorkState.Indexed
+import weco.catalogue.internal_model.work.{Work, WorkState}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait SingleWorkDirectives extends CustomDirectives {
+  implicit val ec: ExecutionContext
+
+  implicit class RouteOps(
+    work: Future[Either[ElasticsearchError, Work[WorkState.Indexed]]]) {
+    def mapVisible(
+      f: Work.Visible[WorkState.Indexed] => Future[Route]): Future[Route] =
+      work.map {
+        case Right(work: Work.Visible[Indexed]) =>
+          withFuture(f(work))
+        case Right(work: Work.Redirected[Indexed]) =>
```

Comment:
What happens when a redirected work is accessed from the items api? Is a redirection issued? Where?

Reply:
It redirects you to the appropriate endpoint on the items API.

e.g. suppose A redirects to B. Then:

```
GET /works/{idA}/items
302 Found -> /works/{idB}/items

GET /works/{idB}/items
200 OK -> items on work B
```

I've added a test to show this behaviour.